htdocs/product/class/api_products.class.php

```diff
@@ -263,9 +263,13 @@ function delete($id)
      *
      * @url GET {id}/categories
      */
-    function getCategories($id, $sortfield = "s.rowid", $sortorder = 'ASC', $limit = 0, $page = 0) {
-        $categories = new Categories();
-        return $categories->getListForItem($sortfield, $sortorder, $limit, $page, 'product', $id);
+	function getCategories($id, $sortfield = "s.rowid", $sortorder = 'ASC', $limit = 0, $page = 0)
+	{
+		require_once DOL_DOCUMENT_ROOT.'/categories/class/api_categories.class.php';
```

Comment:
Why this require.
Do you mean  require_once DOL_DOCUMENT_ROOT.'/categories/class/categories.class.php';   instead ?

Reply:
@eldy 
the function "getListForItem" is in "api_categories.class.php"... i move this function in categorie.class.php ?

Reply:
@eldy 
if i move the function "getListForItem" in "categorie.class.php" i can't clean the data with "_cleanObjectDatas" !

Reply:
@eldy 
ok it's fixed with new "getListForItem" function in categorie class